pkg/component/etcd/lease/lease_test.go

```diff
@@ -0,0 +1,268 @@
+			})
+		})
+
+		Context("when deployment is changed from 1 -> 3 replicas", func() {
+			BeforeEach(func() {
+				// Create existing leases
+				for _, l := range []coordinationv1.Lease{
+					memberLease(etcd, 0),
+				} {
+					lease := l
+					Expect(c.Create(ctx, &lease)).To(Succeed())
+				}
+			})
+
+			It("should successfully deploy", func() {
+				Expect(leaseDeployer.Deploy(ctx)).To(Succeed())
+
+				checkMemberLeases(ctx, c, etcd)
+				checkSnapshotLeases(ctx, c, etcd, values)
+			})
+		})
+
+		Context("when backup is disabled", func() {
+			BeforeEach(func() {
+				leaseDeployer = New(c, namespace, Values{
+					BackupEnabled: false,
+					EtcdName:      name,
+					EtcdUID:       uid,
+					Replicas:      replicas,
+				})
+			})
+
+			It("should successfully deploy", func() {
+				// Snapshot leases might exist before, so create them here.
+				for _, l := range []coordinationv1.Lease{
+					{ObjectMeta: metav1.ObjectMeta{Name: values.DeltaSnapshotLeaseName}},
+					{ObjectMeta: metav1.ObjectMeta{Name: values.FullSnapshotLeaseName}},
+				} {
+					lease := l
+					Expect(c.Create(ctx, &lease)).To(Succeed())
+				}
+
+				Expect(leaseDeployer.Deploy(ctx)).To(Succeed())
+
+				checkMemberLeases(ctx, c, etcd)
+
+				_, err := getSnapshotLease(ctx, c, namespace, values.DeltaSnapshotLeaseName)
+				Expect(err).To(matchers.BeNotFoundError())
+				_, err = getSnapshotLease(ctx, c, namespace, values.FullSnapshotLeaseName)
+				Expect(err).To(matchers.BeNotFoundError())
+			})
+		})
+	})
+
+	Describe("#Destroy", func() {
+		Context("when no leases exist", func() {
+			It("should destroy without errors", func() {
+				Expect(leaseDeployer.Destroy(ctx)).To(Succeed())
+			})
+		})
+
+		Context("when leases exist", func() {
+			It("should destroy without errors", func() {
+				for _, l := range []coordinationv1.Lease{
+					memberLease(etcd, 0),
+					memberLease(etcd, 1),
+					memberLease(etcd, 2),
+					{ObjectMeta: metav1.ObjectMeta{Name: values.DeltaSnapshotLeaseName}},
+					{ObjectMeta: metav1.ObjectMeta{Name: values.FullSnapshotLeaseName}},
+				} {
+					lease := l
+					Expect(c.Create(ctx, &lease)).To(Succeed())
+				}
+
+				Expect(leaseDeployer.Destroy(ctx)).To(Succeed())
+
+				_, err := getSnapshotLease(ctx, c, namespace, values.DeltaSnapshotLeaseName)
+				Expect(err).To(matchers.BeNotFoundError())
+				_, err = getSnapshotLease(ctx, c, namespace, values.FullSnapshotLeaseName)
+				Expect(err).To(matchers.BeNotFoundError())
+
+				leases := &coordinationv1.LeaseList{}
+				Expect(c.List(ctx, leases, client.InNamespace(etcd.Namespace), client.MatchingLabels(map[string]string{
+					common.GardenerOwnedBy:           etcd.Name,
+					v1beta1constants.GardenerPurpose: "etcd-member-lease",
+				}))).To(Succeed())
+
+				Expect(leases.Items).To(BeEmpty())
+			})
+		})
+	})
+})
+
+func getSnapshotLease(ctx context.Context, c client.Client, namespace, name string) (*coordinationv1.Lease, error) {
+	fullLease := coordinationv1.Lease{}
```

Comment:
Nit
Change var used in this func to `snapshotLease` as it's general purpose
```suggestion
	snapshotLease := coordinationv1.Lease{}
```

Reply:
Makes sense, I changed it.